testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java

```diff
@@ -1,65 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.trino.tests.product.teradata;
-
-import io.trino.tempto.ProductTest;
-import org.testng.annotations.Test;
-
-import java.sql.Date;
-import java.sql.Timestamp;
-import java.time.LocalDateTime;
-
-import static io.trino.tempto.assertions.QueryAssert.Row.row;
-import static io.trino.tests.product.TestGroups.FUNCTIONS;
-import static io.trino.tests.product.utils.QueryExecutors.onTrino;
-import static org.assertj.core.api.Assertions.assertThat;
-
-public class TestTeradataFunctions
-        extends ProductTest
-{
-    @Test(groups = FUNCTIONS)
-    public void testIndex()
-    {
-        assertThat(onTrino().executeQuery("SELECT index('high', 'ig')")).contains(row(2));
```

Comment:
The advantage of this over a unit test was

- ensuring the td functions are part of the production bundle.
  - for plugins, we manage to violate this many times over, so it's a lesson learned
- ensuring the td functions work without `trino-main` on the classpath
  - for functions, we managed to make them work in unit tests only. I remember a case where one of the maintainers fixed plugin-provided aggregate functions 3 times, because we lacked product test coverage

I think we should keep at least one product tests' test case

Reply:
What setups do these tests run with? Are they executed with all the combinations of connectors and environments?

Reply:
i hope not! Once should be enough (and i guess this was the case)

Reply:
based on the past experience i think we should have at least one test case showing that plugin-provided functions work when loaded in a real server, with real plugin classloader
https://github.com/trinodb/trino/issues/14486 (https://github.com/trinodb/trino/pull/14488)
https://github.com/trinodb/trino/issues/10341 (https://github.com/trinodb/trino/pull/10346, https://github.com/trinodb/trino/pull/10408)